extensions/ql-vscode/src/vscode-tests/no-workspace/astViewer.test.ts

```diff
@@ -15,19 +15,25 @@ const expect = chai.expect;
 
 describe('AstViewer', () => {
   let astRoots: AstItem[];
-  let viewer: AstViewer;
+  let viewer: AstViewer | undefined;
+  let sandbox: sinon.SinonSandbox;
   beforeEach(async () => {
+    sandbox = sinon.createSandbox();
```

Comment:
Use `sinon.sandbox` where possible. This ensures all stubs and mocks are cleaned up when `sandbox.restore()` is called. One problem with the integration tests earlier on was that mocks were not being properly restored and we were seeing confusing errors across tests.